flytekit/types/structured/structured_dataset.py

```diff
@@ -542,7 +551,7 @@ def _get_dataset_type(self, t: typing.Union[Type[StructuredDataset], typing.Any]
             raise ValueError(f"Unrecognized Annotated type for StructuredDataset {t}")
 
         # 2. Fill in columns by checking for StructuredDataset metadata. For example, StructuredDataset[my_cols, parquet]
-        elif issubclass(t, StructuredDataset):
+        elif inspect.isclass(t) and issubclass(t, StructuredDataset):
```

Comment:
remind me again what this `inspect.isclass` is supposed to catch?  can you add a comment?  i keep forgetting.

Reply:
That's for `Annotated[pd.Dataframe, my_col]`. I just moved `expected_python_type = get_args(expected_python_type)[0]` to the beginning of the `to_python` and `to_literal`. Therefore, we don't need `inspect.isclass(t)` any more, so I removed it.